src/standard/x-styling.html

```diff
@@ -144,9 +144,18 @@
             _scopeSelector: this._scopeSelector,
             _styleProperties: this._styleProperties
           };
-          scopeData.key.customStyle = {};
-          this.mixin(scopeData.key.customStyle, this.customStyle);
-          scope._styleCache.store(this.is, info, scopeData.key, this._styles);
+          // the scope cache does not evaluate if @media rules, :host(), or :host-context() rules defined in this element have changed
+          // therefore, if we detect those rules, we opt-out of the scope cache
+          var scopeCacheable = this._styles.every(function(s) {
+            return s._cacheable;
+          });
+          if (scopeCacheable) {
```

Comment:
Ideally, the call to [scope._styleCache.retrieve on line 112](https://github.com/Polymer/polymer/pull/3702/files#diff-7b0b0bbe811e51dd18a55fc34fa53052R112) is also gated on `scopeCacheable`